erpnext/accounts/doctype/journal_entry/journal_entry.js

```diff
@@ -116,7 +116,7 @@ erpnext.accounts.JournalEntry = frappe.ui.form.Controller.extend({
 			if(jvd.reference_type==="Expense Claim") {
 				return {
 					filters: {
-						'approval_status': 'Approved',
+						'workflow_state': 'Approved',
```

Comment:
there will be no workflow_state field in Expense Claim by default

Reply:
Was waiting for Leave Application to get merged. Will fix now.